mono/metadata/icall-def.h

```diff
@@ -529,10 +529,8 @@ NOHANDLES(ICALL(LINUXNETWORKCHANGE_2, "CreateNLSocket", ves_icall_System_Net_Net
 NOHANDLES(ICALL(LINUXNETWORKCHANGE_3, "ReadEvents", ves_icall_System_Net_NetworkInformation_LinuxNetworkChange_ReadEvents))
 #endif
 
-#if !defined(DISABLE_SOCKETS) && (defined(HOST_DARWIN) || defined(HOST_BSD))
```

Comment:
don't we need to keep the `DISABLE_SOCKETS`? not sure if this will cause issues on iOS/tvOS/watchOS.